lib/utility/script_scanner.rb

```diff
@@ -0,0 +1,66 @@
+require 'utility/imsdb_parser'
+
+module Utility
+	class ScriptScanner
+
+		def populate_all_scripts
+			Movie.all.each do |movie|
+				scan_script(movie.title)
+			end
+		end
+
+		def get_script(movie)
+			return NoMovieExists unless movie
+
+			parser = Utility::ImsdbParser.new
+			parser.populate_script(movie)
+		end
+
+		def	scan_script(movie)
+			get_script(movie) if movie.is_scraped == false
+
+			words = movie.script.split(' ')
+			words.each do |word|
+				movie.ingredients << Ingredient.find_by(name: word) unless Ingredient.find_by(name: word).nil?
```

Comment:
Two lookups per action here again. You could halve the time of these actions by just storing it in a variable first